### PR TITLE
Embedding extraction

### PR DESCRIPTION
**Pull request by The-Mats**
Please review #1 first!

The goal is to allow the use of models to extract embeddings that are then used to train a classifier:
- [x] Fix tensorflow, torch incompatibility errors
- [x] Make sure Perch works
- [x] Add BirdNET model
- [x] Implement mechanic to extract embeddings and use new classifier head
- [x] Fix BirdNET errors after merge
- [x] Test all embedding experiments and a birdset experiment
- [x] Integrate models folder
- [x] Reorder embedding experiments

### CODE REVIEW

File: birdset/datamodule/beans_datamodule.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        dataset = dataset.remove_columns('Unnamed: 0')
+
+        # Then we have to map the label to integers if they are strings (One-hot encoding)
+        labels = set()
+        for split in dataset.keys():
+            labels.update(dataset[split]["labels"])
+
+        label_to_id = {lbl: i for i, lbl in enumerate(labels)}
+
+        def label_to_id_hot(row):
```
Comment (raphaelschwinger) [inline birdset/datamodule/beans_datamodule.py]: @XgamerTV Take a look at the `_classes_one_hot` function of the `BaseDataModuleHF' ;)

File: birdset/modules/embedding_module.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+class EmbeddingModuleConfig(NetworkConfig):
+    """
+    A dataclass that makes sure the model inherits from EmbeddingClassifier.
+
+    """
+    model: Union[EmbeddingModel, Module] = None # Model for extracting the embeddings
+
+class EmbeddingModule(BaseModule):
+    """
+    MultilabelModule is a PyTorch Lightning module for multilabel classification tasks.
```
Comment (raphaelschwinger) [inline birdset/modules/embedding_module.py]:
```suggestion
    EmbeddingModule is a PyTorch Lightning module for multilabel classification tasks.
    The Module expects a embedding_model that will be used to extract embeddings and than trains a network with these embeddings as input features.
```

File: birdset/modules/models/linear_classifier.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    ) -> None:
+        """
+        Initialize the LinearClassifier.
+        
+        Args:
+            num_classes (int): The number of output classes for the classifier.
+            in_features (int): The size of the input for the linear classifier.
+        """
+        super().__init__()
+        self.classifier = nn.Linear(in_features, num_classes)
```
Comment (raphaelschwinger) [inline birdset/modules/models/linear_classifier.py:20]: @XgamerTV This is now a 2 layer NN, why did you choose 2 layers instead of one?

File: configs/module/network/birdnet_inference.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 model: 
   _target_: birdset.modules.models.birdnet.BirdNetModel
-  num_classes: ${datamodule.dataset.n_classes}
```
Comment (raphaelschwinger) [inline configs/module/network/birdnet_inference.yaml]: @XgamerTV Could we revert this change or does this than break something? Haven't we concluded that this will be overridden in train.py anyway? But there might be use cases why the line was like it is.

File: scripts/beans.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (raphaelschwinger) [inline scripts/beans.sh:1]: @XgamerTV Could you add some descriptions of the added experiments in the README.md?

**Review by raphaelschwinger — APPROVED** [review]
@XgamerTV Great work, thanks!

File: birdset/modules/models/linear_classifier.py (hunk 3 above)
Comment (The-Mats) [reply birdset/modules/models/linear_classifier.py:20]: I thought it would be one layer from the embedding suze to the class amount how can I use one?

File: configs/module/network/birdnet_inference.yaml (hunk 4 above)
Comment (The-Mats) [reply configs/module/network/birdnet_inference.yaml]: What is the change I don't understand the diagram, but my experiments don't use birdnet_inference

File: birdset/modules/models/linear_classifier.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        '''self.classifier = nn.Sequential(
+            nn.Linear(in_features, 128),
+            nn.ReLU(),
+            nn.Dropout(0.5),
+            nn.Linear(128, 64),
+            nn.ReLU(),
+            nn.Linear(64, num_classes),
+        )'''
```
Comment (raphaelschwinger) [inline birdset/modules/models/linear_classifier.py]:
```suggestion

```

**Review by raphaelschwinger — APPROVED** [review]

File: configs/module/network/birdnet_inference.yaml (hunk 4 above)
Comment (raphaelschwinger) [reply configs/module/network/birdnet_inference.yaml]: @XgamerTV You deleted the line:
```
num_classes: ${datamodule.dataset.n_classes}
```
and added the line:
```
num_classes: null
```
Instead.

File: configs/module/network/birdnet_inference.yaml [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
+  num_classes: null
```
Comment (raphaelschwinger) [inline configs/module/network/birdnet_inference.yaml]:
```suggestion
  num_classes: ${datamodule.dataset.n_classes}
```
